From b93735854be32153c20de746889a00ce4d5169db Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Thu, 5 Mar 2015 17:35:29 +0530
Subject: [PATCH] opsregion: Add V3P3SX rail in CRC ops-region driver

The patch adds V3P3SX to the CRC PMIC operation region
driver. This resolves the suspend failures in touch/SD
and camera drivers on CHT RVP.

The BIOS already exposes this rail in the OPS region table.
If the ops regios driver doesnt have the entry for this rail,
all slave drivers using this rail will go to an UNKNOWN
acpi power state causing a suspend abort.

Tracked-On: https://jira01.devtools.intel.com/browse/GMIN-3679

Change-Id: I1026d7a9e7b1aa25c91c1d275226325ccf6638db
Signed-off-by: Kumar P, Mahesh <mahesh.kumar.p@intel.com>
Signed-off-by: Iyer, Yegnesh S <yegnesh.s.iyer@intel.com>
---
 drivers/acpi/pmic_opregion/intel_soc_pmic_crc_opregion.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/acpi/pmic_opregion/intel_soc_pmic_crc_opregion.c b/drivers/acpi/pmic_opregion/intel_soc_pmic_crc_opregion.c
index 7ba8141..d8f249a 100644
--- a/drivers/acpi/pmic_opregion/intel_soc_pmic_crc_opregion.c
+++ b/drivers/acpi/pmic_opregion/intel_soc_pmic_crc_opregion.c
@@ -151,6 +151,13 @@ static struct pmic_pwr_table pwr_table[] = {
 			.bit = 0x00,
 		},
 	},/* V105 -> V1P05S, L2 SRAM */
+	{
+		.address = 0x68,
+		.pwr_reg = {
+			.reg = 0x5F,
+			.bit = 0x00,
+		},
+	},/* V3P3SX */
 };
 
 static struct pmic_dptf_table dptf_table[] = {
-- 
1.9.1

